app/src/main/java/org/fossasia/openevent/app/data/AbstractObservable.java

```diff
@@ -18,15 +19,19 @@
 public final class AbstractObservable {
 
     private final ConnectionStatus connectionStatus;
+    private final RateLimiter<String> rateLimiter = new RateLimiter<>(Duration.ofMinutes(10));
 
     public static final class AbstractObservableBuilder<T> {
         private final ConnectionStatus connectionStatus;
+        private final RateLimiter<String> rateLimiter;
         private boolean reload;
+        private String rateLimiterKey;
         private Observable<T> diskObservable;
         private Observable<T> networkObservable;
 
-        public AbstractObservableBuilder(ConnectionStatus connectionStatus) {
+        public AbstractObservableBuilder(ConnectionStatus connectionStatus, RateLimiter<String> rateLimiter) {
```

Comment:
@iamareebjamal I am taking the RateLimiter in the constructor and storing it inside the AbstractObservableBuilder. Is it fine ?

Reply:
Will update tests accordingly if this is fine.

Reply:
This constructor takes connection status just because, there is no reason for this, so, if the rate limiter is taken through the builder methods, this can be avoided

Builders should have no or at most 1 constructor parameter